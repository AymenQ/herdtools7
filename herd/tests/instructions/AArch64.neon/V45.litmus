AArch64 V45
(* Tests STP - store pair, post-indexed *)
{ 
0:X0=x; 0:X1=y;
}

 P0                      ;
 MOVI D0, #255           ;
 MOVI D1, #255           ;
 STP D0, D1, [X0], #-8    ;
 LDR D2, [X0, #8]        ;
 STR D2, [X1]            ;
 forall (x=255 /\ y=255 /\ 0:X0!=x)
