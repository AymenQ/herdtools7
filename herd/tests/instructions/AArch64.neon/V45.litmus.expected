Test V45 Required
States 1
0:X0=x+-8; x=255; y=255;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (x=255 /\ y=255 /\ not (0:X0=x))
Observation V45 Always 1 0
Hash=538798d804954f9135ac0e781a1e92bf

